snuba_sdk/query_visitors.py

```diff
@@ -288,8 +288,23 @@ def _combine(self, query: "query.Query", returns: Mapping[str, None]) -> None:
                 )
 
             for group_exp in non_aggregates:
-                if group_exp not in query.groupby:
-                    raise InvalidQuery(f"{group_exp} missing from the groupby")
+                # Array joins work a little differently. In this case the entire arrayjoin
+                # expression shouldn't be in the groupby, just the column that is being
+                # array joined
+                if (
+                    isinstance(group_exp, Function)
+                    and group_exp.function == "arrayJoin"
+                ):
+                    assert group_exp.parameters and isinstance(
+                        group_exp.parameters[0], Column
```

Comment:
Are you sure we cannot arrayJoin over a complex expression? `arrayJoin(f(x))`? It should still work.

Reply:
This is old. I'll update.